lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -962,6 +997,88 @@ static LogicalResult buildAllocOp(ComponentLoweringState &componentState,
                     IntegerAttr::get(rewriter.getI1Type(), llvm::APInt(1, 1)));
   componentState.registerMemoryInterface(allocOp.getResult(),
                                          calyx::MemoryInterface(memoryOp));
+
+  assert(memtype.getElementTypeBitWidth() <= 64 &&
+         "element bitwidth should not exceed 64");
+  bool isFloat = !memtype.getElementType().isInteger();
+
+  auto shape = allocOp.getType().getShape();
+  int totalSize =
+      std::reduce(shape.begin(), shape.end(), 1, std::multiplies<int>());
+  // The `totalSize <= 1` check is a hack to:
+  // https://github.com/llvm/circt/pull/2661 We should instead never pass
+  // multi-dimensional memories to Calyx, as discussed in:
+  // https://github.com/calyxir/calyx/issues/907
+  if (!(shape.size() <= 1 || totalSize <= 1)) {
+    allocOp.emitError("input memory dimension must be empty or one.");
+    return failure();
+  }
+
+  std::vector<uint64_t> flattenedVals(totalSize, 0);
+
+  if (isa<memref::GetGlobalOp>(allocOp)) {
+    auto getGlobalOp = cast<memref::GetGlobalOp>(allocOp);
+    auto *symbolTableOp =
+        getGlobalOp->template getParentWithTrait<mlir::OpTrait::SymbolTable>();
+    auto globalOp = dyn_cast_or_null<memref::GlobalOp>(
+        SymbolTable::lookupSymbolIn(symbolTableOp, getGlobalOp.getNameAttr()));
+    // Flatten the values in the attribute
+    auto cstAttr = llvm::dyn_cast_or_null<DenseElementsAttr>(
+        globalOp.getConstantInitValue());
+    int sizeCount = 0;
+    for (auto attr : cstAttr.template getValues<Attribute>()) {
+      assert((isa<mlir::FloatAttr>(attr) || isa<mlir::IntegerAttr>(attr)) &&
```

Comment:
> Mistakes happen, I'd personally feel most comfortable with some assertion or error returned, because right now it's UB if it is not an Integer/Float

can you explain what does `UB` stand for? @cgyurgyik 
and here is the assertion right?